libs/libskynet/src/skylinkBitfield.ts

```diff
@@ -0,0 +1,166 @@
+// parseSkylinkBitfield parses a skylink bitfield and returns the corresponding
+// version, offset, and fetchSize.
+function parseSkylinkBitfield(skylink: Uint8Array): [bigint, bigint, bigint, string | null] {
+  // Validate the input.
+  if (skylink.length !== 34) {
+    return [0n, 0n, 0n, "provided skylink has incorrect length"];
+  }
+
+  // Extract the bitfield.
+  let bitfield = new DataView(skylink.buffer).getUint16(0, true);
+
+  // Extract the version.
+  const version = (bitfield & 3) + 1;
+  // Only versions 1 and 2 are recognized.
+  if (version !== 1 && version !== 2) {
+    return [0n, 0n, 0n, "provided skylink has unrecognized version"];
+  }
+
+  // If the skylink is set to version 2, we only recognize the link if
+  // the rest of the bits in the bitfield are empty.
+  if (version === 2) {
+    if ((bitfield & 3) !== bitfield) {
+      return [0n, 0n, 0n, "provided skylink has unrecognized version"];
+    }
+    return [BigInt(version), 0n, 0n, null];
+  }
+
+  // Verify that the mode is valid, then fetch the mode.
+  bitfield = bitfield >> 2;
+  if ((bitfield & 255) === 255) {
+    return [0n, 0n, 0n, "provided skylink has an unrecognized version"];
```

Comment:
It's the same error message because it has the same action item: You provided a skylink that uses a version which was created after the implementation of this function, therefore this function can't process the skylink. Extra information wouldn't be helpful, this error is essentially telling you to upgrade your software.

any `getSkylinkVersion` function would just be an alias to processing the bitfield. The version is decided by the entire construction of the bitfield.